src/cockpit/beiboot.py

```diff
@@ -225,6 +240,22 @@ async def do_custom_command(self, command: str, args: tuple, fds: list[int], std
         if command == 'cockpit.fail-no-cockpit':
             raise CockpitProblem('no-cockpit', message=args[0])
 
+        if command == 'cockpit.check-os-release':
+            remote_os = parse_os_release(args[0])
+            logger.debug("cockpit.check-os-release: remote: %r", remote_os)
+            try:
+                with open("/etc/os-release") as f:
+                    local_os = parse_os_release(f.read())
+            except OSError as e:
+                logger.warning("failed to read local /etc/os-release, skipping OS compatibility check: %s", e)
+                return
+
+            logger.debug("cockpit.check-os-release: local: %r", local_os)
+            # for now, just support the same OS
+            if remote_os.get('ID') != local_os.get('ID') or remote_os.get('VERSION_ID') != local_os.get('VERSION_ID'):
+                msg = f'Unsupported OS {remote_os.get("ID", "?")} {remote_os.get("VERSION_ID", "?")}'
```

Comment:
Do we need to give the user a tiny bit more information? In theory their OS is supported if they install the relevant `cockpit` packages (but then it is no longer beiboot).

Just wondering in general if it makes sense to expand the error message a bit as currently it is not really actionable.

Reply:
Yeah, good point -- the previous error message (i.e. from main) isn't great either. It's not translated/translatable and has the "no-cockpit" error code in it, and doesn't give you anything actionable either. I think we need to handle this error code properly on the login page, and find a way to convey the os-release information in an extra error field. I can work on that, either as a prerequisite (and block this), or as a follow-up.

Reply:
This can be a follow up (as long as it makes next week's release)

Reply:
See #21124. For this PR we should amend the error message with something like "Logging into machines without installed Cockpit is only supported on Fedora 40" (i.e. `$ID $VERSION_ID`). This also easily extends in the future to an itemized list of supported OSes.

@garrett WDYT?

Reply:
Looks like this now:

![image](https://github.com/user-attachments/assets/19197cff-1511-4d53-8373-e0363f7ed916)

Reply:
.. and in the next step it could look like this:

> A compatible version of Cockpit is not installed on HOST. This is only supported for the following operating systems on the target machine:
> - fedora 40
> - fedora 41
> - rhel 10

... and so on.